assets/js/custom/MyProfile.js

```diff
@@ -135,18 +135,66 @@ let MyProfile = function (profile_url, save_username,
     $.get(self.toggle_visibility_url + '/' + id, {}, function (data) {
       let visibilityLockId = $('#visibility-lock-' + id)
       let visibilityLockOpenId = $('#visibility-lock-open-' + id)
+      let programName = $('#program-' + id).find('.program-name').text()
+  
+      let translations = []
```

Comment:
140- 152 is almost duplicated code to 107-116. It would make sense to create and use a function for this functionality.